### PR TITLE
ci: skip coverage for non-go changes

### PR DESCRIPTION
**Pull request by tychoish**
The spirit of my change is that we shouldn't run golang coverage tests unless go source files (outside of the test package, which are generally integration tests).

Any failures uncovered in these cases (e.g. docs, fuzzer/e2e test changes) wouldn't be appropriate to fix in the context of these PRs, and don't provide useful/actionable feedback.

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -1,7 +1,13 @@
 name: Test Coverage
 on:
   pull_request:
+    paths:
+      - "**.go"
```
Comment (creachadair) [inline .github/workflows/coverage.yml:5]: Does the coverage tool already know to skip `_test.go` files?

**Review by creachadair — APPROVED** [review]

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (tychoish) [reply .github/workflows/coverage.yml:5]: this is really "run `go test`" and "coverage" is sort of a poor name for this workflow. 

My goal is that you run the go tests any time any go file changes.

Comment (tychoish) [reply .github/workflows/coverage.yml:5]: this is really "run `go test`" and "coverage" is sort of a poor name for this workflow. 

My goal is that you run the go tests any time any go file changes.

**Comment by tychoish** [comment]
Oh this might be a thing that @cmwaters might be able to fix. I believe that this PR isn't mergeable because the required statuses haven't run, but the whole point of the PR is that for changes like this, they probably shouldn't run.

**Comment by marbar3778** [comment]
so we went down this path before but we decided to have them run but skkip the actual running of tests down below. This allows the requiring. 

```
   - uses: technote-space/get-diff-action@v5
        with:
          PATTERNS: |
            **/**.go
            go.mod
            go.sum
```

**Comment by tychoish** [comment]
ah, then we should extend this pattern so that we don't run go tests for changes to go files within the test directory. I'll make that change later.

**Pull request merged**